manyfews/calculations/gefs.py

```diff
@@ -0,0 +1,201 @@
+    """
+    This function is developed for finder the index of the specific cell in gefs data.
+
+    :param latitudeInfo: the latitude matrix of the grid.
+    :param longitudeInfo: the longtitude matrix of the grid.
+    :param latValue: the latitude of the specific cell
+    :param lonValue: the longitude of the specific cell
+    :return: the index of the cell location.
+    """
+
+    # find the Index of the specific location in gefs data.
+    latsIndex = np.where(latitudeInfo == latValue)
+    lonsIndex = np.where(longitudeInfo == lonValue)
+    index = (latsIndex[0][0], lonsIndex[1][0])
+
+    return index
+
+
+def dataBaseWriter(dt, forecastDays, backDays, latValue, lonValue):
+    """
+    This function is used to save data from gefs file into Database.
+    ( calculations_noaaforecast table).
+
+    :param dt: Specify time-step in days. ( the interval is every 0.125 day(3 hours))
+    :param forecastDays: Number of Days into the future that the forecast is for.
+    :param backDays: the back days number you want to extract (unit: day) (Maximum = 3 days).
+    :param latValue: the latitude of the specific cell.
+                    (the solution is 0.5 degree. range [-90, 90] with 0.5 interval)
+    :param lonValue: the longtidue of the specific cell.
+                    (the solution is 0.5 degree. range [-90, 90] with 0.5 interval)
+    :return: none
+    """
+
+    # calculate the number of time steps
+    # For example: update interval: 0.25 day = 6 h.
+    #              Number of Days into the future that the forecast is for
+    #              16 days = 4 * 16 = 65 loop steps.
+    #  Therefore, the gefs files are:
+    #  geavg.t00z.pgrb2a.0p50.f006 ---> geavg.t00z.pgrb2a.0p50.f384
+
+    loopRange = int(forecastDays / dt)
+    deltaHour = int(24 * dt)
+
+    downloadDate = datetime.utcnow() - timedelta(
+        days=int(backDays)
+    )  # download 3 days back GEFS data.
+    fileDate = downloadDate.strftime("%Y%m%d")
+    date = datetime.astimezone(downloadDate, tz=timezone(timedelta(hours=0)))
+
+    for i in range(loopRange):
+        forcastHour = deltaHour + i * deltaHour
+        gefsData = GEFSdownloader(
+            fileDate=fileDate,
+            forecastHour=forcastHour,
+            latValue=latValue,
+            lonValue=lonValue,
+        )
+        print(
+            forcastHour,
+            gefsData[0],
+            gefsData[1],
+            gefsData[2],
+            gefsData[3],
+            gefsData[4],
+            gefsData[5],
+        )
+        gefsData = NoaaForecast(
+            location=Point(latValue, lonValue),
+            date=date,
+            precipitation=gefsData[5],
+            min_temperature=gefsData[2],
+            max_temperature=gefsData[1],
+            wind_u=gefsData[3],
+            wind_v=gefsData[4],
+            relative_humidity=gefsData[0],
+        )
+
+        gefsData.save()
+
+
+dataBaseWriter(dt=0.25, forecastDays=16, backDays=3, latValue=-80.5, lonValue=175)
```

Comment:
Could you move this line out of the file? I don't think it should be part of the Django app. If it's a way to try out the scripts you could either give instructions on how to run it from the `manage.py` shell, or add it as a task in `tasks.py` so it can be run from the admin interface.

Reply:
We should also have latValue and lonValue as settings for now e.g. in the .env file.